src/topology/subset_properties.lean

```diff
@@ -437,31 +437,47 @@ is_clopen_inter hs (is_clopen_compl ht)
 
 end clopen
 
-section irreducible
+section preirreducible
 
-/-- An irreducible set is one where there is no non-trivial pair of disjoint opens. -/
-def is_irreducible (s : set α) : Prop :=
+/-- An preirreducible set is one where there is no non-trivial pair of disjoint opens. -/
+def is_preirreducible (s : set α) : Prop :=
 ∀ (u v : set α), is_open u → is_open v →
-  (∃ x, x ∈ s ∩ u) → (∃ x, x ∈ s ∩ v) → ∃ x, x ∈ s ∩ (u ∩ v)
+  (s ∩ u).nonempty → (s ∩ v).nonempty → (s ∩ (u ∩ v)).nonempty
+
+/-- An irreducible set is one that is nonempty and
+where there is no non-trivial pair of disjoint opens. -/
+def is_irreducible (s : set α) : Prop :=
+s.nonempty ∧ is_preirreducible s
+
+lemma is_irreducible.nonempty {s : set α} (h : is_irreducible s) :
+  s.nonempty := h.1
+
+lemma is_irreducible.is_preirreducible {s : set α} (h : is_irreducible s) :
+  is_preirreducible s := h.2
 
-theorem is_irreducible_empty : is_irreducible (∅ : set α) :=
+theorem is_preirreducible_empty : is_preirreducible (∅ : set α) :=
 λ _ _ _ _ _ ⟨x, h1, h2⟩, h1.elim
 
 theorem is_irreducible_singleton {x} : is_irreducible ({x} : set α) :=
-λ u v _ _ ⟨y, h1, h2⟩ ⟨z, h3, h4⟩, by rw mem_singleton_iff at h1 h3;
-substs y z; exact ⟨x, or.inl rfl, h2, h4⟩
+⟨singleton_nonempty x,
+ λ u v _ _ ⟨y, h1, h2⟩ ⟨z, h3, h4⟩, by rw mem_singleton_iff at h1 h3;
+ substs y z; exact ⟨x, or.inl rfl, h2, h4⟩⟩
 
-theorem is_irreducible_closure {s : set α} (H : is_irreducible s) :
-  is_irreducible (closure s) :=
+theorem is_preirreducible_closure {s : set α} (H : is_preirreducible s) :
+  is_preirreducible (closure s) :=
 λ u v hu hv ⟨y, hycs, hyu⟩ ⟨z, hzcs, hzv⟩,
 let ⟨p, hpu, hps⟩ := mem_closure_iff.1 hycs u hu hyu in
 let ⟨q, hqv, hqs⟩ := mem_closure_iff.1 hzcs v hv hzv in
 let ⟨r, hrs, hruv⟩ := H u v hu hv ⟨p, hps, hpu⟩ ⟨q, hqs, hqv⟩ in
 ⟨r, subset_closure hrs, hruv⟩
 
-theorem exists_irreducible (s : set α) (H : is_irreducible s) :
-  ∃ t : set α, is_irreducible t ∧ s ⊆ t ∧ ∀ u, is_irreducible u → t ⊆ u → u = t :=
-let ⟨m, hm, hsm, hmm⟩ := zorn.zorn_subset₀ { t : set α | is_irreducible t }
+lemma is_irreducible_closure {s : set α} (h : is_irreducible s) :
+  is_irreducible (closure s) :=
+⟨nonempty_closure h.nonempty, is_preirreducible_closure h.is_preirreducible⟩
+
+theorem exists_preirreducible (s : set α) (H : is_preirreducible s) :
+  ∃ t : set α, is_preirreducible t ∧ s ⊆ t ∧ ∀ u, is_preirreducible u → t ⊆ u → u = t :=
+let ⟨m, hm, hsm, hmm⟩ := zorn.zorn_subset₀ { t : set α | is_preirreducible t }
```

Comment:
I think there are normally no spaces around braces in set definitions.

Reply:
Fixed.